File: clients/swift/src/Services/TagsService.swift
import Foundation

public final class TagsService: Sendable {
    private let client: ApiHttpClient

    init(client: ApiHttpClient) { self.client = client }

    public func list(limit: Int? = nil) async throws -> [Tag] {
        var items: [URLQueryItem] = []
        if let l = limit { items.append(.init(name: "limit", value: String(l))) }
        return try await client.send(.init(method: "GET", path: "/tags", queryItems: items))
    }
}

File: clients/typescript/src/services/TagsService.ts
import type { ListTagsRequest } from "../actions/index.js";
import type { ApiClient, RequestOptions } from "../client.js";
import type { Tag } from "../models/index.js";

/** Resource methods for `/tags`. See https://access.mymind.com/api → Tags. */
export class TagsService {
  constructor(private readonly client: ApiClient) {}

  /** `GET /tags`. */
  async list(
    request?: ListTagsRequest,
    options?: RequestOptions,
  ): Promise<Tag[]> {
    return this.client.send<Tag[]>({
      method: "GET",
      path: "/tags",
      query: { limit: request?.limit },
      signal: options?.signal,
    });
  }
}
